sky/clouds/azure.py

```diff
@@ -265,23 +265,30 @@ def _make(instance_list):
                     instance_type=instance_type,
                     # Setting this to None as Azure doesn't separately bill /
                     # attach the accelerators.  Billed as part of the VM type.
-                    accelerators=None)
+                    accelerators=None,
+                    cpu=None,
+                )
                 resource_list.append(r)
             return resource_list
 
         # Currently, handle a filter on accelerators only.
         accelerators = resources.accelerators
         if accelerators is None:
-            # No requirements to filter, so just return a default VM type.
-            return (_make([Azure.get_default_instance_type()]),
-                    fuzzy_candidate_list)
+            # Return a default instance type with the given number of vCPUs.
+            default_instance_type = Azure.get_default_instance_type(
+                cpu=resources.cpu)
+            if default_instance_type is None:
+                return ([], fuzzy_candidate_list)
+            else:
+                return (_make([default_instance_type]), fuzzy_candidate_list)
```

Comment:
ditto on replacing fuzzy_candidate_list?

Reply:
Changed to [].